src/net_processing.cpp

```diff
@@ -1252,6 +1253,33 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
            (GetBlockProofEquivalentTime(*pindexBestHeader, *pindex, *pindexBestHeader, m_chainparams.GetConsensus()) < STALE_RELAY_AGE_LIMIT);
 }
 
+bool PeerManagerImpl::FetchBlock(const NodeId nodeid, const CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    PeerRef peer = GetPeerRef(nodeid);
+    if (peer == nullptr) return false;
+    uint32_t nFetchFlags = 0;
+    CNodeState* state = State(nodeid);
+    if (state == nullptr) {
+        return false;
+    }
+    const int node_sync_height = state->pindexBestKnownBlock ? state->pindexBestKnownBlock->nHeight : -1;
+    if (pindex->nHeight > node_sync_height) {
+        return false;
+    }
+    if (state->fHaveWitness) {
+        nFetchFlags |= MSG_WITNESS_FLAG;
+    }
+    std::vector<CInv> vInv(1);
+    vInv[0] = CInv(MSG_BLOCK | nFetchFlags, pindex->GetBlockHash());
```

Comment:
Use initializer list ctor:

```suggestion
    std::vector<CInv> vInv{CInv(MSG_BLOCK | nFetchFlags, pindex->GetBlockHash())};
```